sdks/go/client.go

```diff
@@ -0,0 +1,221 @@
+package hatchet
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	v0Client "github.com/hatchet-dev/hatchet/pkg/client"
+	"github.com/hatchet-dev/hatchet/pkg/worker"
+	"github.com/hatchet-dev/hatchet/sdks/go/features"
+)
+
+// Client provides the main interface for interacting with Hatchet.
+type Client struct {
+	v0Client v0Client.Client
+}
+
+// NewClient creates a new Hatchet client.
+// Configuration options can be provided to customize the client behavior.
+func NewClient(opts ...v0Client.ClientOpt) (*Client, error) {
+	v0Client, err := v0Client.New(opts...)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Client{
+		v0Client: v0Client,
+	}, nil
+}
+
+// NewWorker creates a worker that can execute workflows.
+// The worker is configured using functional options.
+func (c *Client) NewWorker(name string, options ...WorkerOption) (*worker.Worker, error) {
+	config := &workerConfig{
+		slots:        100,
+		durableSlots: 1000,
+	}
+
+	for _, opt := range options {
+		opt(config)
+	}
+
+	// Use the higher of slots or durableSlots as the max runs
+	// since durable tasks tend to be longer-running
+	maxRuns := config.slots
+	if config.durableSlots > maxRuns {
+		maxRuns = config.durableSlots
+	}
+
+	workerOpts := []worker.WorkerOpt{
+		worker.WithClient(c.v0Client),
+		worker.WithName(name),
+		worker.WithMaxRuns(maxRuns),
+	}
+
+	if config.logger != nil {
+		workerOpts = append(workerOpts, worker.WithLogger(config.logger))
+	}
+
+	if config.logLevel != "" {
+		workerOpts = append(workerOpts, worker.WithLogLevel(config.logLevel))
+	}
+
+	if config.labels != nil {
+		workerOpts = append(workerOpts, worker.WithLabels(config.labels))
+	}
+
+	w, err := worker.NewWorker(workerOpts...)
+	if err != nil {
+		return nil, err
+	}
+
+	// Register workflows with the worker
+	for _, workflow := range config.workflows {
+		req, regularFns, durableFns, onFailureFn := workflow.Dump()
+		err := w.RegisterWorkflowV1(req)
+		if err != nil {
+			return nil, err
+		}
+
+		// Register regular task functions
+		for _, fn := range regularFns {
+			err = w.RegisterAction(fn.ActionID, func(ctx worker.HatchetContext) (interface{}, error) {
+				return fn.Fn(ctx)
+			})
+			if err != nil {
+				return nil, err
+			}
+		}
+
+		// Register durable task functions
+		for _, fn := range durableFns {
+			err = w.RegisterAction(fn.ActionID, func(ctx worker.HatchetContext) (interface{}, error) {
+				return fn.Fn(ctx)
+			})
+			if err != nil {
+				return nil, err
+			}
+		}
+
+		// Register on failure function if exists
+		if onFailureFn != nil {
+			// The on-failure action ID is derived from the workflow name
+			onFailureActionID := strings.ToLower(fmt.Sprintf("%s:on-failure", req.Name))
```

Comment:
I think we're using an underscore as the separator for this in the other SDKs

Reply:
TS SDK: https://github.com/hatchet-dev/hatchet/blob/main/sdks/typescript/src/v1/client/worker/worker-internal.ts#L909

Python SDK: https://github.com/hatchet-dev/hatchet/blob/main/sdks/python/hatchet_sdk/runnables/workflow.py#L1019

I guess we should consolidate the naming for sure 😅